test/test_utils/__init__.py

```diff
@@ -876,50 +883,57 @@ def parse_canary_images(framework, region):
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/mxnet-inference:{fw_version}-gpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/mxnet-inference:{fw_version}-cpu-{py3_version}",
-            ],
-            "pytorch": [
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/pytorch-training:{fw_version}-gpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/pytorch-training:{fw_version}-cpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/pytorch-inference:{fw_version}-gpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/pytorch-inference:{fw_version}-cpu-{py3_version}",
-            ],
-            # TODO: uncomment once cpu training and inference images become available
-            "huggingface_pytorch": [
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-pytorch-training:{fw_version}-gpu-{py3_version}",
-                # f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-pytorch-training:{fw_version}-cpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-pytorch-inference:{fw_version}-gpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-pytorch-inference:{fw_version}-cpu-{py3_version}",
-            ],
-            "huggingface_tensorflow": [
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-tensorflow-training:{fw_version}-gpu-{py3_version}",
-                # f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-tensorflow-training:{fw_version}-cpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-tensorflow-inference:{fw_version}-gpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/huggingface-tensorflow-inference:{fw_version}-cpu-{py3_version}",
-            ],
-            "autogluon": [
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/autogluon-training:{fw_version}-gpu-{py3_version}",
-                f"{registry}.dkr.ecr.{region}.amazonaws.com/autogluon-training:{fw_version}-cpu-{py3_version}",
-            ],
-        }
-        # E3 Images have an additional "e3" tag to distinguish them from the regular "sagemaker" tag
-        if customer_type == "e3":
-            dlc_images += [f"{img}-e3" for img in images[framework]]
+        if fw_version in pre_populated_py3_version:
+            py3_versions = pre_populated_py3_version[fw_version]
         else:
-            dlc_images += images[framework]
+            ## Keeping it here for legacy releases. Based on the new implementation
+            ## the tags always have py3 version appended to it and hence the 
+            ## get_canary_default_tag_py3_version might not be required. 
+            py3_versions = [get_canary_default_tag_py3_version(framework, fw_version)]
```

Comment:
it seems like there is no case where this will be hit - let's remove rather than keeping it around